tlv/truncated.go

```diff
@@ -40,6 +40,15 @@ func ETUint16(w io.Writer, val interface{}, buf *[8]byte) error {
 	return NewTypeForEncodingErr(val, "uint16")
 }
 
+// ETUint16T is an Encoder for truncated uint16 values, where leading zeros will
+// be omitted. An error is returned if val is not a *uint16.
+func ETUint16T(w io.Writer, val uint16, buf *[8]byte) error {
+	binary.BigEndian.PutUint16(buf[:2], val)
+	numZeros := numLeadingZeroBytes16(val)
+	_, err := w.Write(buf[numZeros:2])
+	return err
+}
```

Comment:
not necessarily, the prior unified record used the non-generic encoder, so this keeps it on par with the prior version. it felt like the proper time to add it for completeness of the library since we will surely use it again in the future, and we have already have rigorous testing on the truncated encodings that we were able to reuse

Reply:
ok